### PR TITLE
Ar 62 en tant quutilisateur je souhaite pouvoir creer une area poc generic component

### PR DESCRIPTION
**Pull request by romainpanno**
* **Checklist before requesting a review**
- [x] The commits messages follows our guidelines
- [x] I have performed a self-review of my code
- [x] I have documented my code
- [ ] Do we need to implement unit tests?

* **In what state is your pull request?**
- [x] Ready to merge / Waiting a reviwer to see my work.
- [ ] Work In Progress (WIP) / My work is not finish but i want daily reviews. (Draft)
- [ ] CI Review / I want to see what the CI thinks of my work.
- [ ] Need feedback / Just to have feedback on what i produced. (May not be merged)

* **What kind of changes does this PR introduce?** (You can choose multiple)
- [ ] Bug fix
- [x] Feature request
- [ ] New / Updated documentation
- [ ] Testing CI ( Make the pull request in draft mode)

* **Describe your changes**

* **Issue ticket number and link**

## Summary by CodeRabbit

- **New Features**
  - Added a new form component to manage and render form data.
  - Implemented a dynamic form structure that adapts to different field configurations.

- **Documentation**
  - Provided clear interface definitions for form field data structures.

### CODE REVIEW

**Review by badidu09 — DISMISSED** [review]

File: front/app/AREAFormTest/page.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    ];
+    const [datas, setDatas] = React.useState<{ [key: string]: string }>({});
+
+    useEffect(() => {
+        if (Object.keys(datas).length !== 0) {
+            console.log("data filled");
+            console.log(datas);
+        } else {
+            console.log("nope");
+        }
```
Comment (Vivelis) [inline front/app/AREAFormTest/page.tsx:24]: Log message should be more explicit.

File: front/app/Components/AREAForm.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+/**
+ * @component AREAForm component
+ * @param {ActionJsonArray} fields - Fields to display
+ * @param setDatas
+ * @return {JSX.Element} AREAForm component with fields to display
+ * @note This component is used to display a form with fields and submit button that will return the data in JSON format
+ */
+const AREAForm: React.FC<{ fields: ActionJsonArray, setDatas: (data: string) => void }> = ({fields, setDatas}) => {
+    const [formData, setFormData] = useState<{ [key: string]: string }>({});
+    const [error, setError] = useState<string | null>(null);
```
Comment (Vivelis) [inline front/app/Components/AREAForm.tsx:18]: Should be no spaces around curly brackets.
```suggestion
const AREAForm: React.FC<{fields: ActionJsonArray, setDatas: (data: string) => void}> = ({fields, setDatas}) => {
    const [formData, setFormData] = useState<{[key: string]: string}>({});
    const [error, setError] = useState<string | null>(null);
```

File: front/app/Interfaces/ActionJson.interface.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+ * @interface ActionJsonArrayType
+ * @description Interface for ActionJsonArray
+ * @extends {Array<ActionJsonInterface>}
+ * @see {ActionJsonInterface}
+ */
+ */
+interface ActionJsonArray extends Array<ActionJsonInterface> {}
+
+export type { ActionJsonInterface, ActionJsonArray };
+
```
Comment (Vivelis) [inline front/app/Interfaces/ActionJson.interface.tsx:26]: Sould be on one empty line at end of file.
```suggestion
export type { ActionJsonInterface, ActionJsonArray };
```

**Review by Vivelis — CHANGES_REQUESTED** [review]

File: front/app/AREAFormTest/page.tsx (hunk 1 above)
Comment (romainpanno) [reply front/app/AREAFormTest/page.tsx:24]: mdrr i forgot to remove sry

Comment (romainpanno) [reply front/app/AREAFormTest/page.tsx:24]: ah yes but it's a test page

Comment (Vivelis) [reply front/app/AREAFormTest/page.tsx:24]: Ok no problem

**Review by Vivelis — APPROVED** [review]

**Review by badidu09 — APPROVED** [review]

**Pull request merged**